src/sysupdate/sysupdate-feature.c

```diff
@@ -0,0 +1,118 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include "conf-parser.h"
+#include "hash-funcs.h"
+#include "path-util.h"
+#include "sysupdate-feature.h"
+#include "sysupdate.h"
+#include "web-util.h"
+
+static Feature *feature_free(Feature *f) {
+        free(f->id);
+
+        free(f->description);
+        free(f->documentation);
+        free(f->appstream);
+
+        return mfree(f);
+}
+
+Feature *feature_new(void) {
+        Feature *f;
+
+        f = new0(Feature, 1);
```

Comment:
`new0` → `new`

(after all you use structured initialization right after, hence no need to zero-initialize things)